tests/test_date.py

```diff
@@ -297,12 +469,115 @@ def test_should_parse_with_no_break_space_in_dates(self):
+            except ValueError as error:
+                result = error
+            finally:
+                self.results.append(result)
+
+    def given_multiple_dates(self, date_to_parse):
+        self.date_strings = date_to_parse.keys()
+        self.dates = list(date_to_parse.values())
+
+    def given_known_languages(self, *shortnames):
+        self.ordered_languages = OrderedDict()
+
+        self.known_languages = [self.language_map.get(shortname)
+                                for shortname in shortnames]
+
+        for language in self.known_languages:
+            self.ordered_languages[language.shortname] = language
+
+        self.language_loader._data = self.ordered_languages
+
+    def given_empty_languages_parser(self):
+        self._data = {}
+        self.parser = date.DateDataParser(allow_redetect_language=True)
+
+        language_loader = LanguageDataLoader()
+        self.language_map = date.default_language_loader.get_language_map()
+        self.language_loader = language_loader
+        language_loader._load_data = MethodType(self.given_empty_languages_parser, language_loader)
+        self.add_patch(patch('dateparser.date.default_language_loader', new=language_loader))
+
+    def given_date_string(self, date_string):
+        self.date_string = date_string
+
+    def given_date_format(self, date_format):
+        self.date_format = date_format
+
+    def given_expected_date(self, expected_date):
+        self.expected_date = expected_date
+
+    def given_today(self):
+        self.today = datetime.utcnow().replace(hour=0, minute=0, second=0, microsecond=0)
+
+    def given_yesterday(self):
+        self.yesterday = self.today - relativedelta(days=1)
+
+    def given_day_before_yesterday(self):
+        self.day_before_yesterday = self.today - relativedelta(days=2)
+
+    def when_date_data_is_parsed(self):
+        self.date_data = self.parser.get_date_data(self.date_string)
+
+    def when_should_parse_date_with_timezones_using_format(self):
+        self.date_data = self.parser.get_date_data(self.date_string, date_formats=[self.date_format])
+
+    def then_check_time(self):
+        self.assertIsNotNone(self.date_data['date_obj'])
+
+    def then_time_in_today(self):
+        self.assertEqual(self.today.date(), self.date_data['date_obj'].date())
+
+    def then_today_was_parsed(self):
+        self.assertIsNotNone(self.date_data['date_obj'])
+
+    def then_date_was_parsed(self):
+        self.assertEqual(self.today.date(), self.date_data['date_obj'].date())
+
+    def then_check_yesterday(self):
+        self.assertIsNotNone(self.date_data['date_obj'],
+                             "could not parse date from: %s" % self.date_string)
+
+    def then_yesterday_is_parsed(self):
+        self.check_equal(self.yesterday.date(),
+                         self.date_data['date_obj'].date(), self.date_string)
+
+    def then_check_day_before_yesterday(self):
```

Comment:
Again, how is this check different from `then_check_time` or `then_today_was_parsed`?